### ISSUE TITLE
fuzz bug: "[..] which is not in the layout"

### ISSUE THREAD

**Issue by kripken** [Cretonne/cretonne#217]
[w.zip](https://github.com/stoklund/cretonne/files/1623495/w.zip)

**Comment by stoklund** [comment]
This is a bug in the wasm->cton translation. The IL fails verification immediately after translation, ebb9 is never inserted in the layout:

```
inst6: v4 is defined by ebb9 which is not in the layout
inst6: brnz.i32 v7, ebb2(v4)

function u0:0(i64 vmctx) -> i32 native {
                                ebb0(v0: i64):
@000b                               v3 = iconst.i32 1
@000d                               brz v3, ebb14
@0014                               jump ebb8

                                ebb8:
@001e                               trap user0

                                ebb13:
@0028                               v8 = iconst.i32 0
@002a                               jump ebb12(v8)

                                ebb12(v7: i32):
@002b                               brnz v7, ebb2(v4)
@002e                               jump ebb7

                                ebb7:
@002f                               jump ebb6

                                ebb14:
@0031                               jump ebb6

                                ebb6:
@0033                               jump ebb5

                                ebb5:
@0034                               jump ebb4

                                ebb4:
@0035                               jump ebb3

                                ebb3:
@0036                               trap user0

                                ebb2(v2: i32):
@0038                               jump ebb1(v2)

                                ebb1(v1: i32):
@0038                               return v1
}
```

cc @sunfishcode

**Comment by pepyakin** [comment]
Here is a more minimized version:

```WebAssembly
(module
  (func (result i32)
      block (result i32) ;; or loop
        unreachable
      end
      block
      end
      i32.clz
  )
)
```

```
inst1: v2 is defined by ebb2 which is not in the layout
inst1: v3 = clz.i32 v2

function u0:0(i32 vmctx) -> i32 native {
                                ebb0(v0: i32):
@0003                               trap user0

                                ebb3:
@0008                               v3 = clz.i32 v2
@0009                               jump ebb1(v3)

                                ebb1(v1: i32):
@0009                               return v1
}
```

**Comment by pepyakin** [comment]
I guess, since cretonne doesn't emit jump to a next ebb in unreachable code

https://github.com/Cretonne/cretonne/blob/4c317f7326010a2f642571f5ce91ea9fbd7c7668/lib/wasm/src/code_translator.rs#L916-L923

as a result, ebb2 isn't getting emitted.

**Comment by sunfishcode** [comment]
Yeah, wasm's unreachable code rules are pretty bizarre. Fixed in cf8eb45.
